src/Cardano/Wallet/Api/Types.hs

```diff
@@ -202,91 +199,111 @@ getApiMnemonicT (ApiMnemonicT (pw, _)) = pw
+                <> show passphraseMinLength <> " chars"
+        | T.length t > passphraseMaxLength =
+            Left $ TextDecodingError $
+                "passphrase is too long: expected at most "
+                <> show passphraseMaxLength <> " chars"
+        | otherwise =
+            pure $ ApiT $ Passphrase $ BA.convert $ T.encodeUtf8 t
+
+instance ToText (ApiT (Passphrase "encryption")) where
+    toText (ApiT (Passphrase bytes)) = T.decodeUtf8 $ BA.convert bytes
 
 passphraseMinLength :: Int
 passphraseMinLength = 10
 
 passphraseMaxLength :: Int
 passphraseMaxLength = 255
 
-newtype MkApiMnemonicError = MkApiMnemonicError String
-    deriving stock Show
-    deriving newtype Buildable
-
-class MkApiMnemonic sizes purpose where
-    mkApiMnemonic
-        :: [Text] -> Either MkApiMnemonicError (ApiMnemonicT sizes purpose)
-
-newtype DecodeApiWalletNameError
-    = DecodeApiWalletNameError String
-    deriving stock Show
-    deriving newtype Buildable
-
-decodeApiWalletName :: Text -> Either DecodeApiWalletNameError (ApiT WalletName)
-decodeApiWalletName t
-    | T.length t < walletNameMinLength =
-        Left $ DecodeApiWalletNameError $
-            "name is too short: expected at least "
-                <> show walletNameMinLength <> " chars"
-    | T.length t > walletNameMaxLength =
-        Left $ DecodeApiWalletNameError $
-            "name is too long: expected at most "
-                <> show walletNameMaxLength <> " chars"
-    | otherwise =
-        return $ ApiT $ WalletName t
-
-encodeApiWalletName :: ApiT WalletName -> Text
-encodeApiWalletName = getWalletName . getApiT
+instance FromText (ApiT WalletId) where
+    fromText = maybe
+        (Left $ TextDecodingError "A wallet ID must be a valid UUID")
+        (Right . ApiT . WalletId)
+        . UUID.fromText
+
+instance ToText (ApiT WalletId) where
+    toText (ApiT (WalletId i)) = UUID.toText i
+
+instance FromText (ApiT WalletName) where
+    fromText t
+        | T.length t < walletNameMinLength =
+            Left $ TextDecodingError $
+                "name is too short: expected at least "
+                    <> show walletNameMinLength <> " chars"
+        | T.length t > walletNameMaxLength =
+            Left $ TextDecodingError $
+                "name is too long: expected at most "
+                    <> show walletNameMaxLength <> " chars"
+        | otherwise =
+            return $ ApiT $ WalletName t
+
+instance ToText (ApiT WalletName) where
+    toText = getWalletName . getApiT
 
 walletNameMinLength :: Int
 walletNameMinLength = 1
 
 walletNameMaxLength :: Int
 walletNameMaxLength = 255
 
+instance MkApiMnemonic sizes purpose => FromText (ApiMnemonicT sizes purpose)
+  where
+    fromText =
+        left (TextDecodingError . show)
+        . mkApiMnemonic @sizes @purpose
+        . T.words
+
+instance ToText (ApiMnemonicT sizes purpose) where
+    toText (ApiMnemonicT (!_, xs)) = T.unwords xs
```

Comment:
I am not a big fan of this one :no_mouth: We would have to artificially concatenate words in order to use the instance which is quite cumbersome especially since the instance itself here provide very little value compare to the already existing `mkApiMnemonic`. So I'll suggest to simple use that `MkApiMnemonic` constructor available.